zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java

```diff
@@ -231,6 +238,27 @@ private Integer mergeTranslations(
                         .getDocument().getId());
                 textFlowDAO.makePersistent(targetTf);
                 textFlowDAO.flush();
+
+                if (Events.exists() && !localeContentStateMap.isEmpty()) {
+                    for (Map.Entry<Long, ContentState> entry : localeContentStateMap
+                            .entrySet()) {
+                        HTextFlowTarget updatedTarget =
+                                targetTf.getTargets().get(entry.getKey());
+
+                        TextFlowTargetStateEvent event =
+                                new TextFlowTargetStateEvent(null,
+                                        targetVersionId,
+                                        targetTf.getDocument().getId(),
+                                        targetTf.getId(),
+                                        updatedTarget.getLocale().getLocaleId(),
+                                        updatedTarget.getId(),
+                                        updatedTarget.getState(),
+                                        entry.getValue());
+
+                        Events.instance().raiseTransactionSuccessEvent(
+                                TextFlowTargetStateEvent.EVENT_NAME, event);
```

Comment:
I was thinking more of raising one big event with the information of all the changed textflow targets. That way we gain some performance improvements.
This is no worse than what's happening at the moment though, which is a single event per updated text flow. But we can improve it.